vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1145,11 +1145,23 @@ def execute_model(
         else:
             assert intermediate_tensors is not None
             assert self.intermediate_tensors is not None
+            tp = self.vllm_config.parallel_config.tensor_parallel_size
+            enabled_sp = self.vllm_config.compilation_config.pass_config. \
+                enable_sequence_parallelism
+            is_residual_scattered = tp > 1 and enabled_sp \
+                and num_input_tokens % tp == 0
```

Comment:
I think this deserves a comment explaining what's happening here - this is pretty novel stuff

Reply:
Perhaps it could be factored out into a helper function, which would help readers keep track of what's going on as well.